pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_controller.go

```diff
@@ -95,6 +100,21 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req kube_ctrl.Request
 	return kube_ctrl.Result{}, nil
 }
 
+func (r *GatewayReconciler) setKumaOwner(ctx context.Context, gateway *gatewayapi.Gateway) error {
+	gatewayKey := kube_types.NamespacedName{Name: gateway.Name, Namespace: gateway.Namespace}
+	gatewayK8s := &mesh_k8s.Gateway{}
+
+	if err := r.Client.Get(ctx, gatewayKey, gatewayK8s); err != nil {
+		return errors.Wrap(err, "could not find Kuma Gateway Kubernetes object")
+	}
+
+	if err := kube_controllerutil.SetControllerReference(gateway, gatewayK8s, r.Scheme); err != nil {
```

Comment:
Where does the Kuma Gateway change get written back to the API server? I was going to suggest doing this in the `Upsert`, but I don't see a way to set the metadata through the Kuma resource interface ...

Reply:
Yeah I forgot to include the `client.Update` call when I excised this from the other PR. 

Unfortunately this doesn't actually lead to the `gateway.kuma.io` object being deleted, because the _`Mesh`_ owns the object, too, and so an owner reference remains. Gonna look into whether there's a different way to handle this.